src/components/pages/profile/account/SetMFA.vue

```diff
@@ -0,0 +1,371 @@
+interface TotpIdentity {
+  issuer?: string;
+  label?: string;
+}
+
+const { t } = useI18n();
+
+/** Reference to slider component */
+const appSliderCo = ref<InstanceType<typeof AppSlider> | null>(null);
+
+/** Used to show/hide the modal */
+const showMFASettingsModal = ref(false);
+
+/** Is multi factor authentication enabled? */
+const isEnabled = ref(false);
+
+/** Stepper component state */
+const appStepper = ref({
+  selectedStepIndex: 0,
+  steps: [
+    {
+      name: t(`accountPage.mfa.scan.title`),
+      validated: false,
+      error: false,
+    },
+    {
+      name: t(`accountPage.mfa.enterCode.title`),
+      validated: false,
+      error: false,
+    },
+    {
+      name: t(`accountPage.mfa.result.title`),
+      validated: false,
+      error: false,
+    },
+  ] as Step[],
+});
+
+const props = defineProps<{
+  contactId: string;
+}>();
+
+/** Authenticator app url to scan (otpauth://totp/...) */
+const totpUrl = ref<string | undefined>(undefined);
+
+/** Information about the app totp is set up for */
+const totpIdentity = ref<TotpIdentity>({
+  issuer: undefined,
+  label: undefined,
+});
+
+/** Secret used to generate totp */
+const totpSecret = ref(new Secret());
+
+/** User token used to verify totp  */
+const userToken = ref('');
+
+/** Is the user token valid? */
+const userTokenValid = ref(false);
+
+/** TOTP instance */
+let totp: TOTP | null = null;
+
+const modalTitle = computed(() => t(`accountPage.mfa.modalTitle`));
+const codeLabel = computed(() => t(`accountPage.mfa.codeInput.label`));
+
+/** Called when the modal is closed */
+const onCloseMFAModal = () => {
+  closeMFAModal();
+  resetState();
+};
+
+/** Close the modal */
+const closeMFAModal = () => {
+  showMFASettingsModal.value = false;
+};
+
+/** Toggle the modal */
+const toggleMFAModal = () => {
+  showMFASettingsModal.value = !showMFASettingsModal.value;
+};
+
+const saveMFA = async () => {
+  closeMFAModal();
+  const result = await createContactMfa(props.contactId, {
+    secret: totpSecret.value.base32,
+    token: userToken.value,
+    type: ContactMfaType.TOTP,
+  });
+  isEnabled.value = true;
+  resetState();
+  return result;
+};
+
+/**
+ * Disable MFA for the contact
+ * TODO: Show confirmation dialog
+ */
+const disableMFA = async () => {
+  await deleteContactMfa(props.contactId);
+  isEnabled.value = false;
+  resetState();
+};
+
+/** Called when the slider changes */
+const onSlideChange = (details: AppSliderSlideEventDetails) => {
+  syncStepperWithSlider(details);
+  validateStep(details);
+};
+
+/** Reset / init the state of the component */
+const resetState = () => {
+  appSliderCo.value?.toSlide(0);
+  appStepper.value.selectedStepIndex = 0;
+  appStepper.value.steps.forEach((step) => {
+    step.validated = false;
+    step.error = false;
+  });
+  userToken.value = '';
+  userTokenValid.value = false;
+};
+
+/** Sync the stepper with the slider */
+const syncStepperWithSlider = (details: AppSliderSlideEventDetails) => {
```

Comment:
This feels like useful common functionality. Could you move the stepper inside the slider component, maybe with an optional param for steps so you can use the slider without the stepper? I think if it's internal they will also be able to use the same state for active step index so you don't need any syncing code

Reply:
Good idea